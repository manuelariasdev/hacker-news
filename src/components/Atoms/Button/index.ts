export * from './Button'
export * from './buttonStyle'