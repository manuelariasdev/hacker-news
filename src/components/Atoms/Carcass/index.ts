export * from './Carcass'
export * from './carcassStyle'