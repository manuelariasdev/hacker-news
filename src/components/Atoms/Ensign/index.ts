export * from './Ensign'
export * from './ensignStyles'