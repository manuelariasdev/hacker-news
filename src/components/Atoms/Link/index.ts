export * from './Link'
export * from './linkStyle'