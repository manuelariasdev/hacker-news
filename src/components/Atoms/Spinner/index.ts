export * from './Spiner'
export * from './spinerStyles'