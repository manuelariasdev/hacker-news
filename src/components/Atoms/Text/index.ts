export * from './Text'
export * from './textStyle'