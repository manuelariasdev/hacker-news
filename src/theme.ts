
import { DefaultTheme } from 'styled-components'

const theme: DefaultTheme = {
  borderRadius: {
    sm: '3px',
    md: '6px',
    lg: '15px',
    xl: '50px'
  },
  boxShadow: {
    sm: ' 0 1px 4px 0 rgba(0, 21, 41, 0.12)'
  },
  fontSize: {
    xs: '0.68rem', 
    sm: '0.875rem', 
    md: '16px',
    lg: '1.25rem', 
    xl: '1.75rem'
  },
  letterSpacing: {
    md: 0.25
  },
  lineHeight: {
    xs: '1rem',
    sm: '1.25rem',
    md: '16px',
    lg: '1.25rem',
    xl: '1.75rem'
  },
  transition: 'all 700ms cubic-bezier(0.15, 1, 0.20, 1)'
}

export default theme
