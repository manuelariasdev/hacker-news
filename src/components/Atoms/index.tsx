import { Ensign } from './Ensign'
import { Button } from './Button'
import { Icon } from './Icon'
import { IconButton } from './IconButton'
import { Link } from './Link'
import { Carcass } from './Carcass'
import { Spinner } from './Spinner'
import { Text } from './Text'
import { Title } from './Title'


export {
  Ensign,
  Button,
  Icon,
  IconButton,
  Link,
  Carcass,
  Spinner,
  Text,
  Title
}
