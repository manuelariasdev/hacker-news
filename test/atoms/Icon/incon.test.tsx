/**
 * @jest-environment jsdom
 */
 import * as React from "react";
 import { ThemeProvider } from 'styled-components'
 import theme from '../../../theme';
 import { render } from '@testing-library/react'
import { Icon } from "../../../src/components/Atoms/Icon";


describe('Icon', () => { 
    const width = 30;
    const heigth = 30;
    test('debe de hacer match con el snapshot Icon', () => { 
       const container = render(<ThemeProvider theme={theme}><Icon name='angular' width={width} heigth={heigth} /></ThemeProvider>)
        expect( container ).toMatchSnapshot();
     })
 })