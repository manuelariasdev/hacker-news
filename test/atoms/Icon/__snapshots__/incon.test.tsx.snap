// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon debe de hacer match con el snapshot Icon 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        height="30"
        viewBox="0 0 32 32"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 2 2.966 6.648l1.988 17.234L16 30l11.046-6.118 1.988-17.234L16 2z"
          style="fill: #dd0031;"
        />
        <path
          d="M16 2v3.108-.014V30l11.046-6.118 1.988-17.234L16 2z"
          style="fill: #c3002f;"
        />
        <path
          d="m16 5.094-8.148 18.27h3.038l1.638-4.088h6.916l1.638 4.088h3.038L16 5.094Zm2.38 11.662h-4.76L16 11.03Z"
          style="fill: #fff;"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      height="30"
      viewBox="0 0 32 32"
      width="30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 2 2.966 6.648l1.988 17.234L16 30l11.046-6.118 1.988-17.234L16 2z"
        style="fill: #dd0031;"
      />
      <path
        d="M16 2v3.108-.014V30l11.046-6.118 1.988-17.234L16 2z"
        style="fill: #c3002f;"
      />
      <path
        d="m16 5.094-8.148 18.27h3.038l1.638-4.088h6.916l1.638 4.088h3.038L16 5.094Zm2.38 11.662h-4.76L16 11.03Z"
        style="fill: #fff;"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
